plugins/package-managers/node/src/main/kotlin/Npm.kt

```diff
@@ -198,8 +198,11 @@ open class Npm(
 
         // Create packages for all modules found in the workspace and add them to the graph builder. They are
         // reused when they are referenced by scope dependencies.
-        val packages = parseInstalledModules(workingDir)
-        graphBuilder.addPackages(packages)
+        val isWorkspacesProject = (findWorkspaceSubmodules(workingDir) - workingDir).isNotEmpty()
```

Comment:
For the shinkwrap test project, do you know why those now removed packages were actually installed in the first place, if they are not referenced from anywhere?

Reply:
`npm` does not know about the scope excludes.

Reply:
Thanks, I somehow missed that the test asset was specific to testing scope excludes.